wordpress-mobile/WordPress-Android
# Use glide instead of Volley on ReaderPostList screen

Fixes #7916 

Note: 
- #7910 needs to be merged before reviewing this PR as `feature/glide-reader-post-list` was created from the `feature/glide-auth-headers`.
- ReaderThumbnailStrip still uses Volley - I'll create another PR with a fix as this PR is already too big

This PR replaces Volley with Glide on the ReaderPostList screen - featured images, video thumbnails, profile images and site icons are now displayed using Glide.  

This PR also introduces ImageManager - a wrapper around Glide for loading images so we can easily replace glide with another library if we ever need to.

To test:
- make sure all images including images from private resources are loaded correctly in the ReaderPostList
- make sure video thumbnails are loaded correctly in the ReaderPostList
- make sure GIF featured images are being animated
